controller/konnect/reconciler_generic.go

```diff
@@ -177,8 +177,27 @@ func (r *KonnectEntityReconciler[T, TEnt]) Reconcile(
 	// If a type has a KongService ref, handle it.
 	res, err = handleKongServiceRef(ctx, r.Client, ent)
 	if err != nil {
-		if !errors.As(err, &ReferencedKongServiceIsBeingDeleted{}) {
+		switch {
+		// in case the referenced KongService is being deleted, disregard the error
+		// and continue
+		case errors.As(err, &ReferencedKongServiceIsBeingDeleted{}):
+			log.Info(logger, "referenced KongService is being deleted, proceeding with reconciliation", err.Error())
+		case errors.As(err, &ReferencedObjectDoesNotExist{}):
+			if ent.GetDeletionTimestamp().IsZero() {
+				log.Info(logger, "referenced KongService does not exist, proceeding with reconciliation", err.Error())
+			}
+			if controllerutil.RemoveFinalizer(ent, KonnectCleanupFinalizer) {
+				if err := r.Client.Update(ctx, ent); err != nil {
+					if k8serrors.IsConflict(err) {
+						return ctrl.Result{Requeue: true}, nil
```

Comment:
```suggestion
						return ctrl.Result{}, nil
```

<img width="870" height="252" alt="Image" src="https://github.com/user-attachments/assets/4b407d8e-0fd0-4621-8ce3-fab7557e6367" />

WDYT?

Reply:
I've put 1 second, as here we really need to requeue. I conflict happens, we still have to remove the finalizer from the entity.